src/Compilers/Core/Portable/Symbols/ISymbolExtensions.cs

```diff
@@ -1,7 +1,18 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System.Collections.Immutable;
+using System.Threading;
+
 namespace Microsoft.CodeAnalysis
 {
+    public static class CompilationExtensions
+    {
+        public static ImmutableArray<TypeDeclarationInfo> GetTopLevelTypeDeclarationInfos(this Compilation compilation, CancellationToken cancellationToken = default)
```

Comment:
i lke the name TopLevel here.  I'm not sure why it's not the same name as what's on Compilation itself.

Reply:
I have been thinking about whether to have the `TypeDeclarationInfo` to be able to represent nested types as well, so we can have something like `GetTypeDeclarationInfos(bool includeNestedTypes)` if needed in the future. Otherwise, this new type feels too specific to the import-completion. However, I couldn't think of a worthy use case for the additional cost. So at the moment I'm inclined to just make it top level type only (and will change all the names to reflect that).